pkg/test/eventually.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"testing"
+	"time"
+)
+
+// A Condition is a function that returns true when a test condition is satisfied.
+type Condition func() bool
+
+const (
+	defaultInterval = 1 * time.Second
+	defaultDeadline = 30 * time.Second
+)
+
+// DEFAULT is the default EventualOpts instance used for the static test.Eventually function.
+var DEFAULT = NewEventualOps(defaultInterval, defaultDeadline)
+
+// Eventually polls cond until it completes (returns true) or times out (resulting in a test failure).
+func Eventually(t *testing.T, name string, cond Condition) {
+	t.Helper()
+	DEFAULT.Eventually(t, name, cond)
+}
+
+// EventualOpts is defines a polling strategy for operations that must eventually succeed.
+// EventualOpts is thread safe and re-usable: a single instance can be constructed for a test run and reused for many conditions.
+type EventualOpts struct {
+	interval, deadline time.Duration
+}
+
+// Eventual constructs an EventualOpts instance with the provided polling interval and deadline.
+func NewEventualOps(interval, deadline time.Duration) EventualOpts {
```

Comment:
Should this be NewEventualOpts? Or should the return value type be named EventualOps?